ddtrace/sampler.py

```diff
@@ -186,6 +221,7 @@ def __init__(
             rate_limit = int(get_env("trace", "rate_limit", default=self.DEFAULT_RATE_LIMIT))  # type: ignore[arg-type]
 
         # Ensure rules is a list
+        self.rules = []  # type: List[SamplingRule]
```

Comment:
Is this line necessary? Since `rules` has the type `Optional[List[SamplingRule]]` it should either be None or a List. If `rules == None` line 230 sets it to an empty list. I think we can set the typing there (basically add `# type: List[SamplingRule]`) and remove this line.

Reply:
We want to set this as early as possible so mypy can validate any of the places we set `self.rules = value` to ensure we only ever have a `List[SamplingRule]`